gems/pending/util/duplicate_blocker/dedup_handler.rb

```diff
@@ -63,8 +68,8 @@ def handle(method, *args)
       key = key_generator.call(method, *args)
       desc = descriptor.call(method, *args)
 
-      entry = update_history(key, desc)
-      tripped?(entry) ? on_blocking_call(desc) : method[*args]
+      entry = update_history(key.hash, desc)
+      tripped?(entry, desc) ? on_blocking_call(desc) : method[*args]
```

Comment:
That's not a lot of memory at all.  The event catcher maybe runs at 200-300MB right now, so you're talking 0.2-0.3%  I would recommend against it at all, unless we can find a way to make the diff nicer.  Passing the description all through the parameters feels wrong, leading to a [data clump](http://martinfowler.com/bliki/DataClump.html).